v4/newrelic/segments.go

```diff
@@ -4,13 +4,30 @@
 package newrelic
 
 import (
+	"context"
 	"net/http"
+	"sync"
+
+	"go.opentelemetry.io/otel/api/trace"
 )
 
+type span struct {
+	sync.Mutex
+	Span trace.Span
+	ctx  context.Context
+	// TODO: linked list and ballooning memory?
+	parent *span
+	ended  bool
+	// TODO: reference cycles?
+	txn *Transaction
```

Comment:
As far as I know, the Golang garbage collector should do fine with reference cycles. Cycles would be more of a problem for languages who do actual reference counting.